src/Streamly/Internal/Data/Stream/StreamD.hs

```diff
@@ -1077,22 +1077,17 @@ splitParse (Parser pstep initial extract) (Stream step state) =
                             src  = Prelude.reverse src0
                         return $ Skip $ ParseChunksBuf src s buf1 pst1
                     -- XXX Halt 0 common case?
-                    PR.Stop n pst1 -> do
-                        res <- extract pst1
-                        case res of
-                            Left err -> error err
-                            Right a -> do
-                                assert (n <= length (x:buf)) (return ())
-                                let src = Prelude.reverse (Prelude.take n (x:buf))
-                                return $ Skip $ ParseChunksYield a (ParseChunksInit src s)
+                    PR.Stop n b -> do
+                        assert (n <= length (x:buf)) (return ())
+                        let src = Prelude.reverse (Prelude.take n (x:buf))
+                        return $ Skip $
+                            ParseChunksYield b (ParseChunksInit src s)
             Skip s -> return $ Skip $ ParseChunksStream s buf pst
             Stop   -> do
-                res <- extract pst
-                case res of
-                    Left err -> error err
-                    Right a -> do
-                        let src = Prelude.reverse buf
-                        return $ Skip $ ParseChunksYield a (ParseChunksInitLeftOver src)
+                b <- extract pst
```

Comment:
```
Stop   -> do
    b <- extract pst
    let src = Prelude.reverse buf
    return $ Skip $
        ParseChunksYield b (ParseChunksInitLeftOver src)
```
The extract at the end seems inconsistent here. I would assume the parsers that go into `splitParse` are terminating and would give a proper result on either `Yield` or `Stop`. Using `extract` here seems inconsistent.  We could save the most recent `Yield` state, which should be extracted, given the stream ends. I don't think that is the behavior right now.

Reply:
`extract` is implemented by the parser, it knows whether there is a consistent partial result available or not, if the parser is implemented correctly extract is guaranteed to  either return a partial result from the last yield or an error. It is doing exactly what you are expecting it to do.